oe_translation.module

```diff
@@ -2,7 +2,199 @@
+  $job = $job_item->getJob();
+  try {
+    $translator_plugin = \Drupal::service('plugin.manager.tmgmt.translator')->createInstance($job->getTranslator()->getPluginId());
+    if ($translator_plugin instanceof AlterableTranslatorInterface) {
+      $translator_plugin->jobItemFormAlter($form, $form_state);
+    }
+  }
+  catch (PluginNotFoundException $exception) {
+    return;
+  }
+}
+
+/**
+ * Implements hook_entity_type_alter().
+ */
+function oe_translation_entity_type_alter(array &$entity_types) {
+  foreach ($entity_types as $entity_type_id => $entity_type) {
+    if (!$entity_type->hasHandlerClass('moderation')) {
+      continue;
+    }
+
+    $entity_type->setHandlerClass('moderation', TranslationModerationHandler::class);
+  }
+}
+
+/**
+ * Implements hook_tmgmt_translatable_fields_alter().
+ */
+function oe_translation_tmgmt_translatable_fields_alter(ContentEntityInterface $entity, &$translatable_fields) {
+  if (isset($translatable_fields['moderation_state'])) {
+    unset($translatable_fields['moderation_state']);
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_access() for the Local Task Item.
+ */
+function oe_translation_tmgmt_local_task_item_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  $access = AccessResult::neutral();
+
+  /** @var \Drupal\tmgmt\TranslatorInterface[] $translators */
+  $translators = \Drupal::entityTypeManager()->getStorage('tmgmt_translator')->loadMultiple();
+  foreach ($translators as $translator) {
+    $plugin_id = $translator->getPluginId();
+    try {
+      $translator_plugin = \Drupal::service('plugin.manager.tmgmt.translator')->createInstance($plugin_id);
+      if ($translator_plugin instanceof LocalTranslatorInterface) {
+        $access = $access->orIf($translator_plugin->localTaskItemAccess($entity, $operation, $account))->addCacheableDependency($translator);
+      }
+    }
+    catch (PluginNotFoundException $exception) {
+      continue;
+    }
+  }
+
+  return $access->addCacheableDependency(\Drupal::entityTypeManager()->getStorage('tmgmt_translator')->getEntityType()->getListCacheTags());
+}
+
+/**
+ * Implements hook_system_breadcrumb_alter().
+ */
+function oe_translation_system_breadcrumb_alter(Breadcrumb &$breadcrumb, RouteMatchInterface $route_match, array $context) {
+  if ($route_match->getRouteName() !== 'entity.tmgmt_local_task_item.canonical') {
+    return;
+  }
+
+  /** @var \Drupal\tmgmt\TranslatorInterface[] $translators */
+  $translators = \Drupal::entityTypeManager()->getStorage('tmgmt_translator')->loadMultiple();
+  foreach ($translators as $translator) {
+    $plugin_id = $translator->getPluginId();
+    try {
+      $translator_plugin = \Drupal::service('plugin.manager.tmgmt.translator')->createInstance($plugin_id);
+      if ($translator_plugin instanceof LocalTranslatorInterface) {
+        $translator_plugin->localTaskItemBreadcrumbAlter($breadcrumb, $route_match, $context);
```

Comment:
I'm not sure why we allow any translator (potentially) to alter task item breadcrumbs, this seems like it would have very unpredictable results.

Reply:
Why? The reason we ask them is so they can alter the hardcoded TMGMT breadcrumbs that link to jobs and tasks and things we may not want to see. Our translator (`PermissionTranslator`) alters that for tasks which use this translator.